SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/activities/SneakerEntryListActivity.java

```diff
@@ -0,0 +1,176 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import com.cyscorpions.dalejulian.sneakpeek.R;
+import com.cyscorpions.dalejulian.sneakpeek.R.id;
+import com.cyscorpions.dalejulian.sneakpeek.R.layout;
+import com.cyscorpions.dalejulian.sneakpeek.R.menu;
+import com.cyscorpions.dalejulian.sneakpeek.adapters.SneakerAdapter;
+import com.cyscorpions.dalejulian.sneakpeek.models.Sneaker;
+import com.cyscorpions.dalejulian.sneakpeek.models.SneakerCategory;
+import com.cyscorpions.dalejulian.sneakpeek.models.SneakerDirectory;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.ContextMenu;
+import android.view.ContextMenu.ContextMenuInfo;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.AdapterView.AdapterContextMenuInfo;
+import android.widget.AdapterView.OnItemClickListener;
+import android.widget.ListView;
+
+public class SneakerEntryListActivity extends Activity implements OnItemClickListener {
+
+	private ListView mEntryList;
+	private ArrayList<Sneaker> mSneakers;
+	private SneakerAdapter mAdapter;
+
+	public static final String LIST_NAME_EXTRA = "name";
+	public static final String LIST_BRAND_EXTRA = "brand";
+	public static final String LIST_SELLING_EXTRA = "selling";
+	public static final String LIST_RARITY_EXTRA = "rarity";
+	public static final String LIST_CATEGORY_EXTRA = "extra";
+	public static final String LIST_DESC_EXTRA = "desc";
+	public static final String LIST_IMGID_EXTRA = "img";
+	public static final String LIST_TITLE_EXTRA = "title";
+	public static final String LIST_ID_EXTRA = "id";
+	public static final String LIST_CATEGORY_EDIT_EXTRA = "editextra";
+	public static final String LIST_FROMLIST_EXTRA = "fromlist";
+
+	private SneakerCategory mCategory = new SneakerCategory();
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.sneaker_entry_listview);
+
+		mCategory.setName(getIntent().getStringExtra(
+				MainActivity.CATEGORY_EXTRA));
+		setTitle("Sneak Peek: " + mCategory.getName());
+
+		mSneakers = SneakerDirectory.get(getApplicationContext())
+				.getSneakersByCategory(mCategory.getName());
+
+		sortEntriesAlphabetically(mSneakers);
+		mAdapter = new SneakerAdapter(this, R.layout.sneaker_entry_listitem,
+				mSneakers);
+		mEntryList = (ListView) findViewById(R.id.listSneakerEntries);
+		mEntryList.setAdapter(mAdapter);
+		mEntryList.setOnItemClickListener(this);
+
+		registerForContextMenu(mEntryList);
+		getActionBar().setDisplayHomeAsUpEnabled(true);
+	}
+
+	@Override
+	public void onItemClick(AdapterView<?> parent, View view, int position,
+			long id) {
+		Sneaker selectedSneaker = mSneakers.get(position);
```

Comment:
Use the getTag(int id) method here instead of getting from the arraylist. The position variable changes invariably.